src/lib/taggingUtils.ts

```diff
@@ -51,8 +51,27 @@ const taggingUtils = {
     return this.buildTag(tagKeys.flag, flagKeys.appData);
   },
 
-  generateFhirVersionTag(): string {
-    return this.buildTag(tagKeys.fhirVersion, fhirService.getFhirVersion());
+  /**
+   * Generate the tag for the FHIR version
+   *
+   * Note: When generating the tag for a search, the tag
+   *       will include the fallback option
+   * @param fhirVersion The FHIR version that should be added as a tag (Default: SDK FHIR version)
+   * @param isSearch Flag indicating if the tags are used for search or not
+   * @returns The string containing the FHIR version tag
+   */
+  generateFhirVersionTag(fhirVersion = fhirService.getFhirVersion(), isSearch = false): string {
+    const original = this.buildTag(tagKeys.fhirVersion, fhirVersion);
+    if (!isSearch) {
+      return original;
+    }
+    /*
+    Earlier versions of the Android SDK did not escape the dots in fhir versions,
+    so we query for both of this encoding and the current standard one by combining
+    both version into an OR query
+    */
+    const fallback = taggingUtils.buildFallbackTag(tagKeys.fhirVersion, fhirVersion);
+    return `(${original},${fallback})`;
```

Comment:
To be honest - I am not a fan of flags - they make it super hard to understand stuff in a global perspective at least in my eyes. Mostly they are solvable with structural patterns - like in this case. Here you could simply split this method into 2. This also fits Single responsibility of SOLID better.